### ISSUE TITLE
Screen reader does not announce items' position when there is a contextual menu button with sub items group in CommandBar.

### ISSUE THREAD

**Issue by alixiashu** [microsoft/fluentui#16782]
### Environment Information

- **Package version(s)**:  ^7.147.0
- **Browser and OS versions**: Chrome/Edge, Win10

### Describe the issue:

User who relies on screen reader will be affected If the object role, state(s), properties, boundary, name, and description are not programmatically determinable, it would be difficult for the platforms that provide Accessibility APIs, and would in turn affect the screen reader dependent users to determine the object properties.

**Screen reader does not announce items' position when there is a contextual menu button with sub items group in CommandBar.**

Btw, with NVDA, it would not announced the "Expanded/Collapsed" state for contextual buttons with sub-items. Even you only want the screen reader as Narrotor, but pls take a look if you can fix it.

### Please provide a reproduction of the issue in a codepen:

code pen demo: https://codepen.io/alixiashu/pen/oNYXKEB?editors=0010
Repro:
1)  Open NVDA/Narrator;
2）Navigate to “filter” button and activate it and navigate to all the lists items
3）Verify screen reader does not announce the position of the list item

#### Actual behavior:

Screen reader does not announce the position of the list items

#### Expected behavior:

Screen reader should announce the position of the list items

### Documentation describing expected behavior

https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={a12f3779-bdf1-40c2-bbcc-0e314ea94858}

**Comment by joschect** [comment]
@alixiashu Thanks for the submission, I can reproduce with NVDA in edge, this looks like a problem with just section props. If you have some time we would welcome a fix if you want to. Otherwise we'll get this added to the backlog.
